EasyPost/PaymentMethod.cs

```diff
@@ -0,0 +1,135 @@
+        private static async Task<CreditCard> GetPaymentMethodByPriority(Priority priority)
+        {
+            PaymentMethod paymentMethods = await All();
+
+            CreditCard? paymentMethod = null;
+            switch (priority)
+            {
+                case Priority.Primary:
+                    paymentMethod = paymentMethods.primary_payment_method;
+                    break;
+                case Priority.Secondary:
+                    paymentMethod = paymentMethods.secondary_payment_method;
+                    break;
+                default:
+                    break;
+            }
+
+            if (paymentMethod == null)
+            {
+                throw new Exception("The chosen payment method is not a valid method. Please try again.");
+            }
+
+            return paymentMethod;
+        }
+
+        /// <summary>
+        ///     List all payment methods for this account.
+        /// </summary>
+        /// <returns>An EasyPost.PaymentMethod summary object.</returns>
+        /// <exception cref="Exception"></exception>
+        public static async Task<PaymentMethod> All()
+        {
+            Request request = new Request("payment_methods", Method.Get);
+
+            PaymentMethod paymentMethod = await request.Execute<PaymentMethod>();
+
+            if (paymentMethod.id == null)
+            {
+                throw new Exception("Billing has not been setup for this user. Please add a payment method.");
+            }
+
+            return paymentMethod;
+        }
+
+        /// <summary>
+        ///     Delete this credit card.
+        /// </summary>
+        /// <param name="paymentMethodId">ID of the payment method to delete.</param>
+        /// <returns>Whether the request was successful or not.</returns>
+        public static async Task<bool> Delete(string paymentMethodId)
+        {
+            Request request = new Request($"credit_cards/{paymentMethodId}", Method.Delete);
+
+            return await request.Execute();
+        }
+
+        /// <summary>
+        ///     Delete a credit card.
+        /// </summary>
+        /// <param name="priority">Which payment method to delete.</param>
+        /// <returns>Whether the request was successful or not.</returns>
+        public static async Task<bool> Delete(Priority priority)
+        {
+            CreditCard paymentMethod = await GetPaymentMethodByPriority(priority);
+
+            return await Delete(paymentMethod.id);
+        }
+
+        /// <summary>
+        ///     Fund a credit card.
+        /// </summary>
+        /// <param name="amount">Amount to fund.</param>
+        /// <param name="paymentMethodId">ID of payment method to fund.</param>
+        /// <returns>An EasyPost.CreditCardFund instance.</returns>
+        public static async Task<CreditCardFund> Fund(string amount, string paymentMethodId)
```

Comment:
Yeah, I'll admit this was somewhat purposeful since, now that I can see how this was designed, I'm not a huge fan of how this was structured.

But I'll concede.